### PR TITLE
Features/ft 839 acgh retrieval by gene

### PR DESCRIPTION
**Pull request by kretep**

### CODE REVIEW

File: grails-app/domain/org/transmartproject/db/dataquery/highdim/chromoregion/DeChromosomalRegion.groovy [hunk 1]
```diff
@@ -29,6 +29,8 @@ class DeChromosomalRegion implements Region {
         name           column: 'region_name'
         numberOfProbes column: 'num_probes'
         platform       column: 'gpl_id'
+        geneSymbol     column: 'gene_symbol'
+        geneId         column: 'gene_id'
```
Comment (cataphract) [inline grails-app/domain/org/transmartproject/db/dataquery/highdim/chromoregion/DeChromosomalRegion.groovy]: This is the default. This hunk is visual pollution.

File: src/groovy/org/transmartproject/db/dataquery/highdim/chromoregion/RegionRowImpl.groovy [hunk 2]
```diff
@@ -30,6 +30,10 @@ final class RegionRowImpl extends AbstractDataRow implements RegionRow {
 
     Integer getNumberOfProbes() { rowList[6] as Integer }
 
+    String getGeneSymbol() { rowList[7] as String }
+
+    Long getGeneId() { rowList[8] as Long }
```
Comment (cataphract) [inline src/groovy/org/transmartproject/db/dataquery/highdim/chromoregion/RegionRowImpl.groovy]: I suppose this has a related change in core-api?

In any case, we already have an interface for returning the gene (among other things), which we've used in the other types. It's the `BiomarkerRow` (or something like that). `RegionRowImpl` should implement that in addition to `Region` (reading  `Region`'s description, I don't think it's entirely appropriate `Region` should extend that `BiomarkerRow`). This won't allow you to give back the gene id, but no one cares about it.

File: transmart-core-db-tests/test/integration/org/transmartproject/db/dataquery/highdim/acgh/AcghEndToEndRetrievalTests.groovy [hunk 3]
```diff
@@ -91,10 +91,14 @@ class AcghEndToEndRetrievalTests {
...
+                hasProperty('geneId', equalTo(testData.regions[1].geneId)))
         assertThat regionRows[1], allOf(
                 hasSameInterfaceProperties(Region, testData.regions[0], ['platform']),
-                hasProperty('label', equalTo(testData.regions[0].cytoband)))
+                hasProperty('label', equalTo(testData.regions[0].cytoband)),
+                hasProperty('geneSymbol', equalTo(testData.regions[0].geneSymbol)),
+                hasProperty('geneId', equalTo(testData.regions[0].geneId)))
 
         assertThat regionRows[1][assayColumns[1]],
                 hasSameInterfaceProperties(AcghValues, testData.acghData[0])
```
Comment (cataphract) [inline transmart-core-db-tests/test/integration/org/transmartproject/db/dataquery/highdim/acgh/AcghEndToEndRetrievalTests.groovy:102]: You added a new data constraint, but no new test using it.

**Comment by kretep** [comment]
Addressed comments. Please have another look.

File: src/groovy/org/transmartproject/db/dataquery/highdim/acgh/AcghModule.groovy [hunk 4]
```diff
@@ -97,6 +104,8 @@ class AcghModule extends AbstractHighDimensionDataTypeModule {
                 property 'region.start'
                 property 'region.end'
                 property 'region.numberOfProbes'
+                property 'region.geneSymbol'
+                property 'region.geneId'
```
Comment (cataphract) [inline src/groovy/org/transmartproject/db/dataquery/highdim/acgh/AcghModule.groovy]: Do you still need to retrieve `geneId`?

Comment (kretep) [inline src/groovy/org/transmartproject/db/dataquery/highdim/acgh/AcghModule.groovy]: You're right. I removed it.

**Pull request merged**